avm/src/daemon/daemon.rs

```diff
@@ -180,7 +195,98 @@ async fn run_agz_b64(agz_b64: &str) -> DaemonResult<()> {
   Ok(())
 }
 
-async fn get_daemon_props() -> Option<&'static DaemonProperties> {
+fn write_b64_file(file_name: &str, content: &str) -> DaemonResult<()> {
+  let pwd = env::current_dir();
+  match pwd {
+    Ok(pwd) => write(
+      format!("{}/{}", pwd.display(), file_name),
+      base64::decode(content).unwrap(),
+    )?,
+    Err(err) => {
+      let err = format!("{:?}", err);
+      return Err(err.into());
+    }
+  }
+  Ok(())
+}
+
+async fn get_files_b64() -> HashMap<String, String> {
+  let mut files_b64 = HashMap::new();
+  files_b64.insert("Dockerfile".to_string(), get_dockerfile_b64().await);
+  files_b64.insert("app.tar.gz".to_string(), get_app_tar_gz_b64().await);
+  files_b64
+}
+
+fn create_app_tar(content: Option<&String>) -> () {
+  if let Some(content) = content {
+    if let Err(err) = write_b64_file("app.tar.gz", &content) {
+      eprintln!("Could not create app.tar.gz. {:?}", err);
+      std::process::exit(1);
+    }
+  }
+}
+
+async fn generate_token() -> String {
+  let body = json!({
+    "deployConfig": deploy::get_config().await,
+  });
+  post_v1("localDaemonToken", body).await
+}
+
+async fn set_local_daemon_props(local_agz_b64: Option<String>) -> () {
+  let files_b64 = get_files_b64().await;
+  create_app_tar(files_b64.get(&"app.tar.gz".to_string()));
```

Comment:
This should be conditional. It's not part of a regular Alan daemon application, only Anycloud.

Reply:
True, will work on this